### DISCUSSION TITLE
How do I run bash code blocks from a qmd in VSCode?

### DISCUSSION THREAD

**Discussion by gtdoctor** [quarto-dev/quarto-cli#9050] [Q&A]
### Description

I have VSCode installed on a local windows machine, I ssh to a linux server where R, RStudio, Quarto are installed. I am unable to run individual {bash} code blocks from within my qmd files. If I run an {r] codeblock, an interactive R terminal window opens and the code runs. If I render the whole file (using Knitr engine), all code blocks run. Using Rstudio on the same server, both code block types work fine. To illustrate my problem:
````

```{bash}
echo "example1"
```

```{r}
print("example2")
```

````
I cannot run the block to return example1 though example2 works fine.

**Comment by mcanouil** [comment]
You can't via the buttons mostly because it depends on the engine which in your case is `knitr`.
`knitr` thus R is handling the code.

I really don't know how RStudio does this but I assume since a lot of things relies on R in it, it might be why it is working.
I don't think VSCode even know that if it's `knitr` engine then `bash` has to go through it.

You can configure VSCode to have a keyboard shortcut that send the command to the currently active terminal.

Anyhow, you should look if this was already reported and if not open a new issue on the appropriate repository: <https://github.com/quarto-dev/quarto/issues>.

**Reply by cderv** [reply]
I thought support was added 🤔 
* https://github.com/quarto-dev/quarto/issues/4

Maybe something is not working anymore.
